source/Engine.cpp

```diff
@@ -2322,15 +2345,31 @@ void Engine::AddSprites(const Ship &ship)
 {
 	bool hasFighters = ship.PositionFighters();
 	double cloak = ship.Cloaking();
+	bool damageHighlight = Preferences::Has("Damage highlights");
 	bool drawCloaked = (cloak && ship.IsYours());
+	bool drawHeat = ship.Heat() > 0.9;
+	bool drawShield = ship.RecentShieldDamage() > 4.;
+	double shield = sqrt(ship.RecentShieldDamage()/ship.Attributes().Get("shields"));
+	double heat = min((ship.Heat() - HEAT_THRESHHOLD) * HEAT_EFFECT_MULTIPLIER, 1.);
 	auto &itemsToDraw = draw[calcTickTock];
-	auto drawObject = [&itemsToDraw, cloak, drawCloaked](const Body &body) -> void
+	auto drawObject = [&itemsToDraw, cloak, shield, heat, drawCloaked, drawShield, drawHeat, damageHighlight](const Body &body) -> void
 	{
-		// Draw cloaked/cloaking sprites swizzled red, and overlay this solid
-		// sprite with an increasingly transparent "regular" sprite.
+		// Draw cloaked/cloaking sprites swizzled transparent grey.
 		if(drawCloaked)
-			itemsToDraw.AddSwizzled(body, 27);
-		itemsToDraw.Add(body, cloak);
+			itemsToDraw.AddSwizzled(body, 10, 0.7);
+		if(damageHighlight)
+		{
+			// Draw the sprite swizzled red when overheating.
+			if(drawHeat)
+				itemsToDraw.AddSwizzled(body, 30);
+			// Overlay this solid sprite with an increasingly transparent "regular" sprite.
+			itemsToDraw.Add(body, max(cloak, drawHeat ? heat : 0.));
+			// Draw another the sprite scaled up swizzled blue over when the shields are damaged.
+			if(drawShield)
+				itemsToDraw.AddSwizzled(body, 29, max(1-shield, 0.75));
+		}
+		else // We would have missed on this step if damageHighlight was disabled.
+			itemsToDraw.Add(body, cloak);
```

Comment:
We generally avoid inline comments.
```suggestion
		else
		{
			// We would have missed on this step if damageHighlight was disabled.
			itemsToDraw.Add(body, cloak);
		}
```
This would technically work without the braces, but it looks spooky to me that way.